users/script.js

```diff
@@ -264,6 +272,107 @@ async function getParticularUserData(
   }
 }
 
+function addAvailibilityFilterOptions() {
+  const options = [
+    { value: 'none', text: NONE },
+    { value: 'active', text: ACTIVE },
+    { value: 'ooo', text: OOO },
+    { value: 'idle', text: IDLE },
+  ];
+
+  const fragment = document.createDocumentFragment();
+
+  options.forEach((option) => {
+    const optionElement = document.createElement('option');
+    optionElement.value = option.value;
+    optionElement.text = option.text;
+    fragment.appendChild(optionElement);
+  });
+
+  availabilityFilter.appendChild(fragment);
+  availabilityFilter.options[0].disabled = true;
+  availabilityFilter.options[0].selected = true;
+}
+
+async function getUsersStatusData(state) {
+  try {
+    const usersRequest = await makeApiCall(
+      `${RDS_API_USERS}status?state=${state}`,
+    );
+    const currentstate = await usersRequest.json();
+    return currentstate.allUserStatus;
+  } catch (err) {
+    throw new Error(`User list request failed with error: ${err}`);
+  }
+}
+
+async function showUserList(users) {
```

Comment:
why is this an async function

Reply:
sorry there is no any use of async here sorry its my mistake

Reply:
😅 no need of sorry.